File: lua/darts/darts.lua
local Darts = {}

function Darts.score(x, y)
	local hypot = math.sqrt(x ^ 2 + y ^ 2)
	if hypot <= 1 then
		return 10
	elseif hypot <= 5 then
		return 5
	elseif hypot <= 10 then
		return 1
	else
		return 0
	end
end

return Darts

File: php/darts/Darts.php
<?php

declare(strict_types=1);

const INNER = 1;
const MIDDLE = 5;
const OUTER = 10;

function score(float $xAxis, float $yAxis): int
{
	$hyp = hypot($xAxis, $yAxis);
	if ($hyp <= INNER) {
		return 10;
	} elseif ($hyp <= MIDDLE) {
		return 5;
	} elseif ($hyp <= OUTER) {
		return 1;
	}
	return 0;
}
